From c13524f3ea5f6338c3aff1b3b3c3598914a46adc Mon Sep 17 00:00:00 2001
From: Lu Hui <luhux76@gmail.com>
Date: Fri, 9 Jun 2023 21:22:10 +0800
Subject: [PATCH] gnu: libsigrokdecode: Fix build & Upgrade.

* gnu/packages/electronics.scm (libsigrokdecode)[version]: use git version
[source]: switch to git repo.
[native-inputs]: add libtool.
[arguments]<#:phases>: not need replace bootstrap.
---
 gnu/packages/electronics.scm | 81 +++++++++++++++++-------------------
 1 file changed, 39 insertions(+), 42 deletions(-)

diff --git a/gnu/packages/electronics.scm b/gnu/packages/electronics.scm
index a8c07f6d94..980827ae1f 100644
--- a/gnu/packages/electronics.scm
+++ b/gnu/packages/electronics.scm
@@ -5,6 +5,7 @@
 ;;; Copyright © 2021 Efraim Flashner <efraim@flashner.co.il>
 ;;; Copyright © 2021 Leo Famulari <leo@famulari.name>
 ;;; Copyright © 2022, 2023 Maxim Cournoyer <maxim.cournoyer@gmail.com>
+;;; Copyright © 2023 Lu Hui <luhux76@gmail.com>
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -79,49 +80,45 @@ (define-public libserialport
     (license license:lgpl3+)))
 
 (define-public libsigrokdecode
-  (package
-    (name "libsigrokdecode")
-    (version "0.5.3")
-    (source (origin
-              (method url-fetch)
-              (uri (string-append
-                    "http://sigrok.org/download/source/libsigrokdecode/libsigrokdecode-"
-                    version ".tar.gz"))
-              (sha256
-               (base32
-                "1h1zi1kpsgf6j2z8j8hjpv1q7n49i3fhqjn8i178rka3cym18265"))
-              (patches
-               (search-patches "libsigrokdecode-python3.9-fix.patch"))))
-    (outputs '("out" "doc"))
-    (arguments
-     `(#:phases
-       (modify-phases %standard-phases
-         (replace 'bootstrap
-           (lambda _
-             (invoke "autoconf")
-             (invoke "aclocal")
-             (invoke "automake" "-ac")))
-         (add-after 'build 'build-doc
-           (lambda _
-             (invoke "doxygen")
-             #t))
-         (add-after 'install 'install-doc
-           (lambda* (#:key outputs #:allow-other-keys)
-             (copy-recursively "doxy/html-api"
-                               (string-append (assoc-ref outputs "doc")
-                                              "/share/doc/libsigrokdecode"))
-             #t)))))
-    (native-inputs
-     (list check doxygen graphviz pkg-config automake autoconf))
-    ;; libsigrokdecode.pc lists "python" in Requires.private, and "glib" in Requires.
-    (propagated-inputs
-     (list glib python))
-    (build-system gnu-build-system)
-    (home-page "https://www.sigrok.org/wiki/Libsigrokdecode")
-    (synopsis "Library providing (streaming) protocol decoding functionality")
-    (description "Libsigrokdecode is a shared library written in C, which provides
+  (let* ((commit "73cb5461acdbd007f4aa9e81385940fad6607696")
+         (revision "0"))
+    (package
+      (name "libsigrokdecode")
+      (version (git-version "0.5.3" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/sigrokproject/libsigrokdecode")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "1i4jpkhb9yqf1fmbvlzdifj9arkspspffn93q8yh4vq5zr27k085"))))
+      (outputs '("out" "doc"))
+      (arguments
+       `(#:phases
+         (modify-phases %standard-phases
+           (add-after 'build 'build-doc
+             (lambda _
+               (invoke "doxygen")
+               #t))
+           (add-after 'install 'install-doc
+             (lambda* (#:key outputs #:allow-other-keys)
+               (copy-recursively "doxy/html-api"
+                                 (string-append (assoc-ref outputs "doc")
+                                                "/share/doc/libsigrokdecode"))
+               #t)))))
+      (native-inputs
+       (list check doxygen graphviz pkg-config automake autoconf libtool))
+      ;; libsigrokdecode.pc lists "python" in Requires.private, and "glib" in Requires.
+      (propagated-inputs
+       (list glib python))
+      (build-system gnu-build-system)
+      (home-page "https://www.sigrok.org/wiki/Libsigrokdecode")
+      (synopsis "Library providing (streaming) protocol decoding functionality")
+      (description "Libsigrokdecode is a shared library written in C, which provides
 (streaming) protocol decoding functionality.")
-    (license license:gpl3+)))
+      (license license:gpl3+))))
 
 (define-public sigrok-firmware-fx2lafw
   (package
-- 
2.40.1

